src/handlers/v1/markets/[chainId]/[marketId]/collateral-swap/index.ts

```diff
@@ -0,0 +1,150 @@
+      if (!targetCollateral) {
+        throw newHttpError(400, { code: '400.6', message: 'target token is not collateral' });
+      }
+
+      // 1. new balancer flash loan logics and append loan logic
+      const [flashLoanLoanLogic, flashLoanRepayLogic] = apisdk.protocols.balancerv2.newFlashLoanLogicPair([
+        { token: _withdrawalToken, amount: amount },
+      ]);
+      logics.push(flashLoanLoanLogic);
+
+      // 2. new and append paraswap swap token logic
+      const quotation = await apisdk.protocols.paraswapv5.getSwapTokenQuotation(chainId, {
+        input: { token: _withdrawalToken, amount: amount },
+        tokenOut: _targetToken,
+        slippage,
+      });
+      logics.push(apisdk.protocols.paraswapv5.newSwapTokenLogic(quotation));
+
+      // 3. new and append compound v3 supply collateral logic
+      logics.push(
+        apisdk.protocols.compoundv3.newSupplyCollateralLogic({
+          marketId,
+          input: { token: _targetToken, amount: quotation.output.amount },
+          balanceBps: common.BPS_BASE,
+        })
+      );
+
+      // 4. new and append compound v3 withdraw logic
+      logics.push(
+        apisdk.protocols.compoundv3.newWithdrawCollateralLogic({
+          marketId,
+          output: { token: _withdrawalToken, amount: amount },
+        })
+      );
+
+      // 5. append balancer flash loan replay logic
+      logics.push(flashLoanRepayLogic);
+
+      const estimateResult = await apisdk.estimateRouterData({ chainId, account, logics });
+      approvals = estimateResult.approvals;
+
+      // 6. calc target position
+      const withdrawalUSD = new BigNumberJS(amount).times(withdrawalCollateral.assetPrice);
+      const targetUSD = new BigNumberJS(quotation.output.amount).times(targetCollateral.assetPrice);
+      const targetBorrowUSD = new BigNumberJS(borrowUSD);
+      const targetCollateralUSD = new BigNumberJS(collateralUSD).minus(withdrawalUSD).plus(targetUSD);
+      const targetBorrowCapacityUSD = new BigNumberJS(borrowCapacityUSD);
+
+      const targetLiquidationLimit = new BigNumberJS(liquidationLimit).minus(
+        withdrawalUSD
+          .times(withdrawalCollateral.liquidateCollateralFactor)
+          .plus(targetUSD.times(targetCollateral.liquidateCollateralFactor))
+      );
+      const targetLiquidationThreshold = common.formatBigUnit(targetLiquidationLimit.div(targetCollateralUSD), 4);
+      targetPosition = {
+        utilization: calcUtilization(targetBorrowCapacityUSD, targetBorrowUSD),
+        healthRate: calcHealthRate(targetCollateralUSD, targetBorrowUSD, targetLiquidationThreshold),
+        netAPR: calcNetAPR(supplyUSD, supplyAPR, targetCollateralUSD, targetBorrowUSD, borrowAPR),
+        totalDebt: common.formatBigUnit(targetBorrowUSD, 2),
+      };
+    }
+
+    const responseBody: GetCollateralSwapQuotationResponseBody = {
+      quotation: { targetTokenAmount: targetAmount, currentPosition, targetPosition },
```

Comment:
I've just noticed that `targetAmount` is always zero, and this isn't tested in unit test. However, integration tests don't fail when checking the target balance. Could you check why this might be the case?

Reply:
Okay. Will dig in to see what happened. Thanks.

Reply:
Two things here:

1. need to use `quotation.quotation.targetTokenAmount` instead of `quotation.targetTokenAmount` in collateral-swap.ts
2. `expect(A.eq(B))` need `.to.be.true` at the end

Reply:
LGTM